Monet 872 content model template integration tests

mrlaessig opened the pull request:
## Summary

Adds integration tests for the environments template api

## Checklist (check all before merging)

- [X] Both unit and integration tests are passing
- [X] There are no breaking changes
- [ ] ~~Changes are reflected in the documentation~~

marcolink commented:
@mrlaessig 
Without knowing how templates exactly work, I wonder if the static used names actually do work.

- What happens if multiple instances of tests run in parallel?
- What happens a test run throws, and the cleanup function is not called? 
- Does it make sense to use a naming convention for templates, and add a similar logic as we have for the [space cleanup](https://github.com/contentful/integration-test-utils/blob/main/src/space/clean-up-test-spaces.ts)?

mrlaessig commented:
> * What happens if multiple instances of tests run in parallel?

Regarding creation there should be an issue. You can have `n` templates with the same name. Anyways to make them more distinguishable I added a random string for each test run to the name. 
But this was a valid point for the cleanup of environment templates! Each test environment template has now a unique description per test run which is used to filter templates for deletion after a test was executed. That way no parallel test runs should accidentally delete tests of the other test runs. 

> * What happens when a test run throws, and the cleanup function is not called?

Orphan environment test templates are now cleaned up before  the test suite is executed if they are older then 1h

> * Does it make sense to use a naming convention for templates, and add a similar logic as we have for the [space cleanup](https://github.com/contentful/integration-test-utils/blob/main/src/space/clean-up-test-spaces.ts)?

Yes, a [similar approach](https://github.com/contentful/contentful-management.js/pull/1507/commits/e999e065304af66324b9a3c5d06a493e2c601bd6#diff-3b89f96d1a840d06f933bf99913ac4906757a56d4acdef7b2b96c0b956600e26R151) is now in place

marcolink reviewed (approved):
@mrlaessig this looks really great :shipit:

marcolink commented on test/helpers.js:
```diff
@@ -137,3 +141,16 @@ export const generateRandomId = (prefix = 'randomId') => {
+    Date.parse(template.sys.updatedAt) + olderThan < Date.now()
+
+  const cleanUpTemplates = templates.filter(filterTemplate).map((templates) => templates.delete())
+  await Promise.all(cleanUpTemplates)
```
```suggestion
  await Promise.allSettled(cleanUpTemplates)
```
Could be an alternative to not make the tests fail when for some reason a template deletion fails.
